src/agent/src/signal.rs

```diff
@@ -76,11 +76,6 @@ async fn handle_sigchild(logger: Logger, sandbox: Arc<Mutex<Sandbox>>) -> Result
 
             p.exit_code = ret;
             let _ = p.exit_tx.take();
-
-            info!(logger, "notify term to close");
-            // close the socket file to notify readStdio to close terminal specifically
-            // in case this process's terminal has been inherited by its children.
-            p.notify_term_close();
```

Comment:
IIRC I tried this exact same change a couple weeks ago to fix #9071, however in my testing that regressed #1160. Could you add an integration test to `k8s-exec.bats` for #1160 and for your use case?

Also have you tested this to see if it might fix your issue? 5536743361b6a3d789826cad1000d84007bb41b0

Reply:
Thanks for your reply. I have updated the PR. PTAL again.  For container io, kata-agent must not terminate io to avoid loss io msg when the exit code of container init process is watched . When init process of container exited,  the children processes will be reaped by os kernel if the value of shareProcessNamespace  param is false.  That is different from exec io.

Reply:
Ok, I will add integration test in the future. Now, I find that the tokio::sync::Notify is not suitable to handle #1160 . if p. notify_term_close() is called before term_exit_notifier.notified() in function do_read_stream(), term_exit_notifier.notified() will be blocked until notify_term_close() called again. That is to say, #1160 is still exists. Of course, the probability of occurrence is lower.